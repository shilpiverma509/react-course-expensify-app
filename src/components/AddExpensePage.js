import React from 'react';

const AddExpensePage =()=>{
    return(
    <div>
        This is from my add expense component
    </div>
    )};

export default AddExpensePage;    


