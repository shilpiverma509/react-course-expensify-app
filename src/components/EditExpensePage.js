import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense} from '../actions/expenses';
import {removeExpense} from '../actions/expenses';

//Refactor editExpensePAge to class based from classless
//Setup mapDispatchToProps editExpense and removeExpense

export class EditExpensePage extends React.Component{
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  onRemove = ()=>{
    this.props.removeExpense({id:this.props.expense.id})
    this.props.history.push('/');
    }
  render(){
    //debugger;
    return (
      <div>
      <ExpenseForm
        expense={this.props.expense}
        onSubmit={this.onSubmit}
      />
      <button onClick={this.onRemove}>
        Remove
        </button>
    </div>
    )
  }
}

// const EditExpensePage = (props) => {
//   return (
//     <div>
//       <ExpenseForm
//         expense={props.expense}
//         onSubmit={(expense) => {
//           props.dispatch(editExpense(props.expense.id, expense));
//           props.history.push('/');
//         }}
//       />
//       <button onClick={()=>{
//         props.dispatch(removeExpense({id:props.expense.id}))
//         props.history.push('/');
//         }}>
//         Remove
//         </button>
//     </div>
//   );
//};



const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};
const mapDispatchToProps = (dispatch,props)=>({
  editExpense:(id,expense)=>dispatch(editExpense(id, expense)),
  removeExpense:(data)=>dispatch(removeExpense(data))
});


export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);
