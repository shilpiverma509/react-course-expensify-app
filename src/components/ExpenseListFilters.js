import React from 'react';
import { connect } from 'react-redux';
import {setTextFilter,sortByAmount,sortByDate,setStartDate,setEndDate} from '../actions/filters'
import { DateRangePicker } from 'react-dates'

class ExpenseListfilters extends React.Component{
    state={
        calendarFocused: null
    };
    onDatesChange = ({startDate,endDate})=>{
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch((setEndDate(endDate)));
    }
    onFocusChange = (calendarFocused)=>{
        this.setState(()=>({calendarFocused}));
    }
    render(){
        return(
        <div>
            <input type="text" 
            value={this.props.filters.text}
            onChange={(e)=>{
                this.props.dispatch(setTextFilter(e.target.value));
            }}
            />
            <select  value={this.props.filters.sortBy}
            onChange={(e)=>{
                (e.target.value==='date')?this.props.dispatch(sortByDate()):this.props.dispatch(sortByAmount());
            }}
            >
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>
            <DateRangePicker
                startDate={this.props.filters.startDate} 
                startDateId="start" 
                endDate={this.props.filters.endDate} 
                endDateId="end"
                onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                numberOfMonths={1}
                isOutsideRange = {()=>false}
                showClearDates={true}
                />
        </div>
        )
    }
    
}
const mapStateToProps = (state)=>{
    return{
        filters:state.filters
    }
}

export default connect(mapStateToProps)(ExpenseListfilters);