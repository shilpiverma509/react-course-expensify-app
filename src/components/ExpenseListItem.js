import React from 'react';
import {removeExpense} from '../actions/expenses';
import {connect} from 'react-redux';


//destructuring props object
/**
 * const ExpenseListItem = (props)=>{
 * <div>
 *  {props.description}{props.amount}
 * </div>
 * }
 * props here is the individual expense array
 * remove button will dispatch an action when it will get clicked

const ExpenseListItem = (props)=>{
    <div>
        <h3>{props.description}</h3>
        <p>{props.amount} -{props.createdAt}</p>
        <button onClick={()=>{
           props.dispatch(removeExpense({id:props.id}));
        }}>Remove</button>
        
    </div>
} 
* The problem is with {props.id} 
* you can't use ES6 enhanced object literal syntax with object properties. 
* You'd need to do:{id:props.id}
*/
const ExpenseListItem = ({dispatch,id,amount,createdAt,description})=>(
    <div>
        <h3>{description}</h3>
        <p>{amount} -{createdAt}</p>
        <button onClick={()=>{
           dispatch(removeExpense({id}));
        }}>Remove</button>
        
    </div>
);
/*we don't need anything from the states.
The connect() gives ExpenseListItem access to the dispatch prop
which is all it needs.
So here the props came from the ExpenseList,then the Action modifier
was called. and that removes the expenses from the state 
*/
// const mapStatetoProps=(state)=>{
//     return{
//         expenses:state.expenses
//     }
// }

export default connect()(ExpenseListItem);