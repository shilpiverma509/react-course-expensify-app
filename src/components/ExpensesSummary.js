import React from 'react';
import SelectExpenses from '../selectors/SelectExpenses';
import selectExpensesTotal from '../selectors/expenses-total';
import {connect} from 'react-redux';
import numeral from 'numeral';

export const ExpensesSummary= (props)=>{
    const expenseWord = props.expenseCount ===1?'expense':'expenses'
    return (
    <div>
            <p>Viewing {props.expenseCount} {expenseWord} totalling {numeral(props.expensesTotal/100).format('$0,0.00')}</p>
    </div>
    )
    
};
const mapStateToProps = (state)=>{
    const visibleExpenses =SelectExpenses(state.expenses,state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal:selectExpensesTotal(visibleExpenses)
    }
}

export default connect (mapStateToProps)(ExpensesSummary);