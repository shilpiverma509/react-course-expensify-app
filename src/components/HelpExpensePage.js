import React from 'react';

const HelpExpensePage =()=>{
    return(
        <div>
            This is from my help component
       </div>
     )};


 export default HelpExpensePage;    