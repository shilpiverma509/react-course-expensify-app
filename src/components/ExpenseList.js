import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import SelectExpenses from '../selectors/SelectExpenses';

const ExpenseList = (props)=>
(
    <div>
        <h1>ExpenseList</h1>
        {props.expenses.map((expense)=>{
            return <ExpenseListItem key={expense.id} {...expense} />
        })}
    </div>
)
const mapStateToProps = (state)=>{
    return {
        expenses:SelectExpenses(state.expenses,state.filters)
    };
}

export default connect(mapStateToProps)(ExpenseList);

