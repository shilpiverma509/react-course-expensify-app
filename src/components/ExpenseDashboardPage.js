/**
 * goal is to list the expenses on the dashboard page
 * how do we access the store information from react componets
 */

import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
//conects your component to the redux store

const ExpenseDashboardPage =(props)=>(
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
    );

export default ExpenseDashboardPage;