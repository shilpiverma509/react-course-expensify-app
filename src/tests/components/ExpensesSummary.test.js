import {ExpensesSummary} from '../../components/ExpensesSummary';
import {shallow} from 'enzyme';
import React from 'react';


test('should correctly display total of 1 expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount = {1} expenseTotal={235} />);
    expect(wrapper).toMatchSnapshot();
})

test('should correctly display total of multiple expenses',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount = {23} expenseTotal={12345} />);
    expect(wrapper).toMatchSnapshot();
})