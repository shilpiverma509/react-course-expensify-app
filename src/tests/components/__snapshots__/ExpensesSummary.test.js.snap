// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly display total of 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $0.00
  </p>
</div>
`;

exports[`should correctly display total of multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    23
     
    expenses
     totalling 
    $0.00
  </p>
</div>
`;
